src/API/Util/PaymentMethods/Enums/Methods.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace OnPay\API\Util\PaymentMethods\Enums;
+
+final class Methods {
+    const CARD = 'card';
+    const MOBILEPAY_CHECKOUT = 'mobilepay_checkout';
+
+    const ALL_METHODS = [
```

Comment:
I'm not sure about the method here, of having the different methods implemented as the different types of objects.
From the point of view of integrators of the SDK, there's no need for differentiating between acquirers, wallets and so on.

However, the idea of having final classes defined for each method with supported currencies is fine. I think this is the way to go around it.